### ISSUE TITLE
Enabling the Jupyter extension causes high CPU usage

### ISSUE THREAD

**Issue by ivanistheone** [microsoft/vscode-jupyter#12827]

**Comment by stephinity** [comment]
I am having the same problem on Windows 10.

Since recently (~1 week), the Jupyter extension causes very high CPU usage (~40% on my machine).
Opening the Windows task manager shows that after creating some cells in the Jupyter NB, VS Code is creating a number of Python sub-process (and a Console Window Host), each running at ~10% CPU. 

Disabling the Jupyter Extension kills most of the Python sub-processes.

Using VS Code for text files or standalone python files does NOT result in this problem.


## Environment data

Version: 1.75.1 (user setup)
Commit: 441438abd1ac652551dbe4d408dfcec8a499b8bf
Date: 2023-02-08T21:32:34.589Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No

Python Extension: v2023.2.0
Jupyter Extension: v2023.1.2010391206


## Steps to reproduce:

1. Open VS Code with Python & Jupyter extensions activated.
2. Create blank Jupyter Notebook
3. Add some Python code to some cells (simple import statements, math like 5*8, and run those cells)
4. Check windows task manager and see new Python subprocesses within VS Code using lots of CPU %


## Example

Screenshot of Windows Task Manager

![image](https://user-images.githubusercontent.com/1645299/218961808-1e058b2a-a28d-4b00-9fec-ba4bca76ef26.png)

**Comment by DonJayamanne** [comment]
Can you disable the python extension, reload vscode and let me know if you run into the same perf issue

@ivanistheone please could you also try the same, please disable the python extension and reload vscode and try this again with just Jupyter enabled

**Comment by stephinity** [comment]
I have tried the following:
- Open VS Code, disable Python extension (which automatically disables all Jupyter extensions also), then exit VS Code
- Open VS Code again, enable only Jupyter extension
- Create new Jupyter Notebook (at this point no CPU increase) but no interpreter is selected
- Try to select interpreter, and get following error msgs

![image](https://user-images.githubusercontent.com/1645299/218988780-2c0cd3c4-c9d6-405a-8d91-56aedb8e6d05.png)

![image](https://user-images.githubusercontent.com/1645299/218988808-120685e6-cf73-42e7-88c4-5f9cdf820ac5.png)

Is it possible to run Jupyter NB without the Python extension?
My Python is installed via Anaconda installation...

**Comment by ivanistheone** [comment]
> please could you also try the same, please disable the python extension and reload vscode and try this again with just Jupyter enabled

Just tested and the problem doesn't occur when only Jupyter extension is active with Python/Pylance off.

So I guess it must be some kind of interaction between the two.

**Comment by ivanistheone** [comment]
In case this is helpful, here is the output of `lsof` for the process that has the CPU pegged at 200%:

<img width="1041" alt="Screen Shot 2023-02-16 at 10 56 31 PM" src="https://user-images.githubusercontent.com/163966/219545909-3758c054-459e-4ccd-a000-573200cda11f.png">

I looked at each of the log files and nothing interesting seems to be happening in any of them, but maybe one of you will recognize a file that should be being accessed?

Is there a way to "inspect" a running process to see what code is currently running?

**Comment by stephinity** [comment]
In case this info is helpful:

I am still having the high CPU problem, and today ran the "Extension Bisect" utility in VS Code.
After going through the different combinations of active extensions, the CPU is high when Jupyter & Pylance extensions are both active.
Having the Jupyter Extension active _with Pylance disabled_ also works fine and at normal CPU.

**Comment by mehulrastogi** [comment]
I also noticed this, and the leading cause might be that somehow a background jupyter notebook is being started, which in my case, also blocks ports. I ran the following and saw new jupyter notebooks being activated as soon as a workspace with python code was launched in vs code
```ps -ef | grep python```

**Comment by DonJayamanne** [comment]
@stephinity @mehulrastogi @ivanistheone 
Please could you go into your settings and enable the following setting
This disables the automatic startup of the jupyter server (for some machines)
Please note, having this feature disabled will impact the performance of starting and running notebooks for the first time in a workspace (i.e. it will be slow)

If enabling this setting has no effect on CPU, then please do undo the change.
Please let me know how this goes.
![Screenshot 2023-02-22 at 05 32 40](https://user-images.githubusercontent.com/1948812/220430183-79edc624-20cd-4565-99b0-d8a856ac25bc.png)

**Comment by ivanistheone** [comment]
@DonJayamanne Nope; I checked that box to disable auto start but the high CPU issue persisted.

<img width="528" alt="Screen Shot 2023-02-21 at 5 59 51 PM" src="https://user-images.githubusercontent.com/163966/220478214-8a7c4d4a-7f1f-4798-8351-6ef3baac8024.png">

**Comment by DonJayamanne** [comment]
@ivanistheone 

Can you try disabling the pylance extension and see if that makes any difference.

If not, then 
* Please could you first close all files (including python files)
* Follow the instructions here https://github.com/microsoft/vscode/wiki/Performance-Issues#profile-the-running-extensions
* Then open a python file to replicate the issue and ensure you capture the per files when CPU peaks

Please share the CPU profile when done.

**Comment by stephinity** [comment]
@DonJayamanne 
I checked the box (Disable Jupyter Auto Start) but it makes no difference to CPU usage, even after reloading VS Code.

Only after disabling Pylance extension (and then reload VS Code) does CPU usage go down again.
